src/containerapp/azext_containerapp/containerapp_env_telemetry_decorator.py

```diff
@@ -0,0 +1,153 @@
+from typing import Any, Dict
+
+DATA_DOG_DEST = 'dataDog'
+APP_INSIGHTS_DEST = 'appInsights'
+logger = get_logger(__name__)
+
+
+class ContainerappEnvTelemetryDataDogPreviewSetDecorator(BaseResource):
+
+    def __init__(self, cmd: AzCliCommand, client: Any, raw_parameters: Dict, models: str):
+        super().__init__(cmd, client, raw_parameters, models)
+        self.managed_env_def = {}
+        self.existing_managed_env_def = {}
+
+    def get_argument_open_telemetry_data_dog_site(self):
+        return self.get_param("site")
+    
+    def get_argument_open_telemetry_data_dog_key(self):
+        return self.get_param("key")
+
+    def get_argument_enable_open_telemetry_traces(self):
+        return self.get_param("enable_open_telemetry_traces")
+    
+    def get_argument_enable_open_telemetry_metrics(self):
+        return self.get_param("enable_open_telemetry_metrics")
+
+    def construct_payload(self):
+        # Get current containerapp env telemetry properties
+        try:
+            self.existing_managed_env_def = self.client.show(cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(), name=self.get_argument_name())
+        except Exception as e:
+            handle_raw_exception(e)
+
+        self.set_up_open_telemetry_data_dog_site()
+        self.set_up_open_telemetry_data_dog_key()
+        self.set_up_open_telemetry_traces_destinations()
+        self.set_up_open_telemetry_metrics_destinations()
+
+    def set_up_open_telemetry_data_dog_site(self):
+        if self.get_argument_open_telemetry_data_dog_site() and self.get_argument_open_telemetry_data_dog_site() is not None:
+            safe_set(self.managed_env_def, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "dataDogConfiguration", "site", value=self.get_argument_open_telemetry_data_dog_site())
+    
+    def set_up_open_telemetry_data_dog_key(self):
+        if self.get_argument_open_telemetry_data_dog_key() and self.get_argument_open_telemetry_data_dog_key() is not None:
+            safe_set(self.managed_env_def, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "dataDogConfiguration", "key", value=self.get_argument_open_telemetry_data_dog_key())
+
+    def set_up_open_telemetry_traces_destinations(self):
+        if self.get_argument_enable_open_telemetry_traces() is not None:
+            existing_traces = safe_get(self.existing_managed_env_def, "properties", "openTelemetryConfiguration", "tracesConfiguration", "destinations")
+            if existing_traces is None:
+                if self.get_argument_enable_open_telemetry_traces():
+                    existing_traces = [DATA_DOG_DEST]
+            else:
+                if self.get_argument_enable_open_telemetry_traces() and DATA_DOG_DEST not in existing_traces:
+                    existing_traces.append(DATA_DOG_DEST) 
+                elif not self.get_argument_enable_open_telemetry_traces() and DATA_DOG_DEST in existing_traces:
```

Comment:
>not self.get_argument_enable_open_telemetry_traces() [](http://example.com/codeflow?start=21&length=52)

when get_argument_enable_open_telemetry_traces is None or "", you will remove it. Is it expected

Reply:
only will hit this logic when self.get_argument_enable_open_telemetry_metrics() is not None